docs/docs/api-reference/starlark-reference/service-config.md

```diff
@@ -117,6 +117,37 @@ config = ServiceConfig(
             wait = "4s"
         ),
     },
+    
+    # The determinsitic public ports that Kurtosis will expose from the container to the machine
+    # This only applies to Docker; the port ids here must be a subset of `ports`
+    # If no ports are provided; normal ports behavior will happen and the exposed ports will be randomly allocated
+    # This doesn't work on Kubernetes!!!
+    # OPTIONAL (Default: {})
+    public_ports = {
+        "grpc": PortSpec(
+            # The port number which we want to expose
+            # MANDATORY
+            number = 3000,
+
+            # Transport protocol for the port (can be either "TCP" or "UDP")
+            # Optional (DEFAULT:"TCP")
+            transport_protocol = "TCP",
+
+            # Application protocol for the port
+            # Optional
+            application_protocol = "http",
+
+            # Kurtosis will automatically perform a check to ensure all declared UDP and TCP ports are open and ready for traffic and connections upon startup.
+            # You may specify a custom wait timeout duration or disable the feature entirely.
+            # You may specify a custom wait timeout duration with a string:
+            #  wait = "2m"
+            # Or, you can disable this feature by setting the value to None:
+            #  wait = None
+            # The feature is enabled by default with a default timeout of 15s
+            # OPTIONAL (DEFAULT:"15s")
+            wait = "4s",
```

Comment:
I'm not sure if this works for the public ports, I think we control just the private ones, but this field is on the `Kurtosis type`